### PR TITLE
Add GA support for confidentialInstanceType

### PR DESCRIPTION
**Pull request by deeglaze**
Finalizes hashicorp/terraform-provider-google#14768
Fixes b/345255719

```release-note:new-resource
compute: added `confidential_instance_config.confidential_instance_type` field to `google_compute_instance`,  `google_compute_instance_template`, and `google_compute_region_instance_template` resources
```

### CODE REVIEW

**Comment by deeglaze** [comment]
Changed to draft since the GA json change hasn't propagated to the Golang derived API package yet.

**Comment by c2thorn** [comment]
> Changed to draft since the GA json change hasn't propagated to the Golang derived API package yet.

please ping me when this is ready for review

**Comment by deeglaze** [comment]
@c2thorn the TestReadPlannedAssetsCoverage  failure appears to be unrelated. The rest is good to go I think.

File: mmv1/third_party/terraform/go.mod.erb [hunk 1]
```diff
@@ -28,18 +28,18 @@ require (
 	github.com/sirupsen/logrus v1.8.1
 	go4.org/netipx v0.0.0-20231129151722-fdeea329fbba
 	golang.org/x/exp v0.0.0-20240409090435-93d18d7e34b8
-	golang.org/x/net v0.25.0
-	golang.org/x/oauth2 v0.20.0
-	google.golang.org/api v0.180.0
```
Comment (c2thorn) [inline mmv1/third_party/terraform/go.mod.erb:33]: is this upgrade needed to support the field in GA?

Comment (deeglaze) [reply mmv1/third_party/terraform/go.mod.erb:33]: It is. The ConfidentialInstanceType field was added to the GA API very recently.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance.go.erb [hunk 2]
```diff
@@ -1025,13 +1025,6 @@ be from 0 to 999,999,999 inclusive.`,
 				Description: `The Confidential VM config being used by the instance.  on_host_maintenance has to be set to TERMINATE or this will fail to create.`,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						<% if version == "ga" -%>
-						"enable_confidential_compute": {
```
[the remark this answers is not in the document]
Comment (deeglaze) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance.go.erb:1029]: I'm not removing it. The else condition is just becoming active. Is the `Required: True` what needs to remain until a major release can fully deprecate it? I've added a deprecation message in another commit at least.

[the remark this answers is not in the document]
Comment (c2thorn) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance.go.erb:1029]: Ahh, I misread the block. Got it!
No, it is not a breaking change to remove a requirement (only the reverse)

**Comment by c2thorn** [comment]
This is a very version-specific change, so I'm going to run tests in our separate GA environment as well just to be sure. If everything's good I'll come back and merge

**Comment by deeglaze** [comment]
Thanks @c2thorn!

**Comment by c2thorn** [comment]
Tests look good, failures were not relevant. LGTM

**Review by c2thorn — APPROVED** [review]

**Pull request merged**